sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/pythonLogicalOperators.scala

```diff
@@ -38,3 +38,30 @@ case class FlatMapGroupsInPandas(
    */
   override val producedAttributes = AttributeSet(output)
 }
+
+trait BaseEvalPython extends UnaryNode {
+
+  def udfs: Seq[PythonUDF]
+
+  def resultAttrs: Seq[Attribute]
+
+  override def output: Seq[Attribute] = child.output ++ resultAttrs
+
+  override def references: AttributeSet = AttributeSet(udfs.flatMap(_.references))
```

Comment:
to work with the `ColumnPruning` rule, we must correctly implement the `references` method. `resultAttrs` are definitely not references.

Reply:
If `references` only cover references in `udfs`, will some output attributes from child that aren't referred by `udfs` be pruned from `BaseEvalPython`?

Reply:
Yea, and this is "column pruning".